frontend/src/components/db/QueryPageTopStack/QueryPageTopStack.tsx

```diff
@@ -0,0 +1,29 @@
+import * as React from "react";
+import { useResultPageType } from "@components/hooks/useResultPageType";
+
+import { ButtonsRenderer } from "./ButtonsRenderer";
+import { DbFileButtons } from "./DbFileButtons";
+import { QueryPageTopStackFrame } from "./QueryPageTopStackFrame";
+import { SearchButtons } from "./SearchButtons";
+
+export const QueryPageTopStack = ({ matches = 0 }: { matches?: number }) => {
+  const { isSearchPage } = useResultPageType();
+
+  if (isSearchPage) {
```

Comment:
we could just do inline conditional rendering on line 16 instead of this top-level `if`